spark-connect/client/src/main/scala-spark-master/io/delta/connect/tables/DeltaTable.scala

```diff
@@ -62,6 +64,98 @@ class DeltaTable private[tables](
    * @since 4.0.0
    */
   def toDF: Dataset[Row] = df
+
+  private def executeHistory(limit: Option[Int]): DataFrame = {
+    val describeHistory = proto.DescribeHistory
+      .newBuilder()
+      .setTable(table)
+    val relation = proto.DeltaRelation.newBuilder().setDescribeHistory(describeHistory).build()
+    val extension = com.google.protobuf.Any.pack(relation)
+    val sparkRelation = spark_proto.Relation.newBuilder().setExtension(extension).build()
+    val df = sparkSession.newDataFrame(_.mergeFrom(sparkRelation))
+    limit match {
+      case Some(limit) => df.limit(limit)
+      case None => df
+    }
+  }
+
+  /**
+   * Get the information of the latest `limit` commits on this table as a Spark DataFrame.
+   * The information is in reverse chronological order.
+   *
+   * @param limit The number of previous commands to get history for
+   * @since 4.0.0
+   */
+  def history(limit: Int): DataFrame = {
+    executeHistory(Some(limit))
+  }
+
+  /**
+   * Get the information available commits on this table as a Spark DataFrame.
+   * The information is in reverse chronological order.
+   *
+   * @since 4.0.0
+   */
+  def history(): DataFrame = {
+    executeHistory(limit = None)
+  }
+
+  /**
+   * :: Evolving ::
+   *
+   * Get the details of a Delta table such as the format, name, and size.
+   *
+   * @since 4.0.0
+   */
+  @Evolving
+  def detail(): DataFrame = {
+    val describeDetail = proto.DescribeDetail
+      .newBuilder()
+      .setTable(table)
+    val relation = proto.DeltaRelation.newBuilder().setDescribeDetail(describeDetail).build()
+    val extension = com.google.protobuf.Any.pack(relation)
+    val sparkRelation = spark_proto.Relation.newBuilder().setExtension(extension).build()
+    sparkSession.newDataFrame(_.mergeFrom(sparkRelation))
+  }
+
+  private def executeRestore(version: Option[Long], timestamp: Option[String]): DataFrame = {
```

Comment:
@andreaschat-db What do you mean? This is a helper function for this `history()` functions which they have doc in explaining what they do, users cannot access `execute...()`

https://github.com/delta-io/delta/blob/51ecfe5ef1b759e88d7e799b94c1919c4aa9eef0/spark/src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala#L51

Reply:
Add a doc explaining what you just explained?

Reply:
@andreaschat-db I deliberately did not add this because it would create duplications of the below `history()`, and I had a review in the past where I was advised to not to do this.

In order for me to understand better to decide the next time, do you have some good reasons for this?